app/models/checkin.rb

```diff
@@ -3,8 +3,9 @@ class Checkin < ActiveRecord::Base
   belongs_to :device
 
   reverse_geocoded_by :lat, :lng do |obj,results|
-    if geo = results.first
-      obj.address = results.first.formatted_address
+    if results.first
+      geo = results.first
+      obj.address = geo.formatted_address
```

Comment:
Have a look to see if you can automatically match the results vars with the obj vars

Not saying it's possible/a good idea, but it might be